File: output/CodeLlama/avatar/Python/Java/codeforces_567_A.java
import java.util.Scanner;

public class codeforces_567_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        for (int i = 0; i < n; i++) {
            if (i == 0) {
                System.out.print(arr[i + 1] - arr[i] + " " + arr[n - 1] - arr[i] + " ");
                continue;
            } else if (i == n - 1) {
                System.out.print(arr[n - 1] - arr[n - 2] + " " + arr[n - 1] - arr[0] + " ");
                continue;
            } else if (arr[i] - arr[i - 1] > arr[i + 1] - arr[i]) {
                System.out.print(arr[i + 1] - arr[i] + " ");
            } else {
                System.out.print(arr[i] - arr[i - 1] + " ");
            }
            if (arr[n - 1] - arr[i] > arr[i] - arr[0]) {
                System.out.print(arr[n - 1] - arr[i] + " ");
            } else {
                System.out.print(arr[i] - arr[0] + " ");
            }
        }
    }
}

File: output/CodeGen/avatar/Java/Python/codeforces_567_A.py
import java.util.Scanner ; import javax.swing.JOptionPane ; 
def main ( ):
	s = Scanner ( System.in )
	a = s.nextInt ( )
	ar = [ ]
	for i in range ( a ):
		ar.append ( s.nextInt ( ) )
	for i in range ( a ):
		max = max ( ar [ i ] - ar [ 0 ], ar [ a - 1 ] - ar [ i ] )
		if i == 0:
			min = ar [ i + 1 ] - ar [ i ]
		else:
			if i == a - 1:
				min = ar [ i ] - ar [ i - 1 ]
			else:
				min = min ( ar [ i ] - ar [ i - 1 ], ar [ i + 1 ] - ar [ i ] )
		print ( min + " " + max )